info_encoding.md

```diff
@@ -1,116 +1,108 @@
+
+For the purpose of this document, common byte order is defined as little endian. In pseudocode, the type `uint16` has the method `encode`, that returns a string containing the common byte order encoding of the number:
+```
+method encode() of uint16 {
+	var out [2]byte
+
+	out[0] = byte(self & 0xff)
+	out[1] = byte(self >> 8)
+
+	return out
+}
+```
+
+For convenience, the function `encodeU16` takes any positive integer as input, converts it to `uint16` and returns the result of `encode`:
+```
+function encodeU16(i int) {
+	assert i >= 0
+	return uint16(i).encode()
+}
+```
 
 ## Requirements
 
-We require the format to support lists of buffers.
-This is enough for most simple operations, and can be extended to
-key-value stores.
+We require the format to support lists of buffers. This is enough for most simple operations, and can be extended to key-value stores.
+A property of many structured data encodings is schemalessness. To get a simpler encoding, we do not require this property. Therefore, derivation steps need to define a schema for the encoding of the info field. For the core derivations, this is given. Custom applications that are defined in extensions need to provide this as well.
 
-To get a simpler encoding, we do not require schemalessness.
-Therefore, derivation steps need to define a schema for the encoding
-of the info field.
-For the core derivations, this is given.
-Custom applications need to provide this as well.
-Note that schemalessness would not help in this context, because 
-all users that are derive a key are on the same page what that key
-is supposed to be used for.
-If they are not, the adversary got them to execute their code already,
-which means that particular user is in trouble anyways (REPHRASE).
+ Note that schemalessness would not help in this context, because all users who derive a key agree, by following the specification, on what that key is supposed to be used for up front, so even in a schema
 
-We require the format to be able to handle buffers that are less than 2<sup>16</sup>B in size.
+We require the format to be able to handle buffers that are at most 65535 bytes in size.
 
-At this point, we do not require to be able to encode nested data.
-However, we will later propose two options for backwards-compatible
-changes to this proposal that support that.
+At this point, we do not require to be able to encode nested data. However, we will later propose two options for backwards-compatible changes to this proposal that support that.
 
 ## Format
 
-The proposed format is called _shallow length prefixed (SLP)_ encoding.
-Shallow, because it does not specify nesting.
-Length-prefixed, because each element is prefixed with its length.
+The proposed format is called _shallow length prefixed (SLP)_ encoding. Shallow, because it does not specify nesting. Length-prefixed, because each element is prefixed with its length.
```

Comment:
it's not a **proposed** format, this isn't a scientific paper, we're actually building things.

i.e. remove "proposed"